### PR TITLE
(feat): show descriptions for params/creds at gen time

### PR DESCRIPTION
**Pull request by schristoff**

### CODE REVIEW

**Comment by schristoff** [comment]
As going through this I found the `Hint:` and `Strategy:` fields in Secret.Source. This would be the optimal way to implement this. I'm going to see if we can figure out how much wiring is needed to get those together.

File: pkg/storage/parameterset_test.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 func TestNewParameterSet(t *testing.T) {
-	ps := NewParameterSet("dev", "myparams",
+	ps := NewParameterSet("dev", "myparams", "",
```
Comment (kichristensen) [inline pkg/storage/parameterset_test.go:18]: Wouldn't it make sense assert that the description is added correctly like this test does with the other inputs?

File: pkg/storage/credentialset_test.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ import (
 )
 
 func TestNewCredentialSet(t *testing.T) {
-	cs := NewCredentialSet("dev", "mycreds", secrets.SourceMap{
+	cs := NewCredentialSet("dev", "mycreds", "", secrets.SourceMap{
```
Comment (kichristensen) [inline pkg/storage/credentialset_test.go:20]: Wouldn't it make sense assert that the description is added correctly like this test does with the other inputs?

File: pkg/storage/parameterset_test.go (hunk 1 above)
Comment (schristoff) [reply pkg/storage/parameterset_test.go:18]: Correct, I'll update these

**Comment by schristoff** [comment]
closing for #3112

**Pull request closed without merging**